src/edu/csus/ecs/pc2/clics/API202306/ProblemService.java

```diff
@@ -0,0 +1,143 @@
+// Copyright (C) 1989-2024 PC2 Development Team: John Clevenger, Douglas Lane, Samir Ashoo, and Troy Boudreau.
+package edu.csus.ecs.pc2.clics.API202306;
+
+import java.util.ArrayList;
+
+import javax.inject.Singleton;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Feature;
+import javax.ws.rs.core.FeatureContext;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+import javax.ws.rs.core.SecurityContext;
+import javax.ws.rs.ext.Provider;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import edu.csus.ecs.pc2.core.IInternalController;
+import edu.csus.ecs.pc2.core.model.IInternalContest;
+import edu.csus.ecs.pc2.core.model.Problem;
+import edu.csus.ecs.pc2.core.util.IJSONTool;
+import edu.csus.ecs.pc2.services.core.JSONUtilities;
+import edu.csus.ecs.pc2.services.eventFeed.WebServer;
+
+/**
+ * WebService to handle problems
+ *
+ * @author John Buck
+ *
+ */
+@Path("/contests/{contestId}/problems")
+@Produces(MediaType.APPLICATION_JSON)
+@Provider
+@Singleton
+public class ProblemService implements Feature {
+
+    private IInternalContest model;
+
+    @SuppressWarnings("unused")
+    private IInternalController controller;
+
+    private JSONTool jsonTool;
+
+    public ProblemService(IInternalContest inContest, IInternalController inController) {
+        super();
+        this.model = inContest;
+        this.controller = inController;
+        jsonTool = new JSONTool(inContest, inController);
+    }
+
+    /**
+     * This method returns a representation of the current contest problems in JSON format. The returned value is a JSON array with one problems description per array element, compying with 2023-06
+     *
+     * @param sc User information
+     * @param contestId The contest
+     * @return a {@link Response} object containing the contest problems in JSON form
+     */
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getProblems(@Context SecurityContext sc, @PathParam("contestId") String contestId) {
+
+        // check contest id
+        if(contestId.equals(model.getContestIdentifier()) == false) {
+            return Response.status(Response.Status.NOT_FOUND).build();
+        }
+        // get the problems from the contest
+        Problem[] problems = model.getProblems();
+        int ord = 1;
+        ArrayList<CLICSProblem> problist = new ArrayList<CLICSProblem>();
+
+        // public only gets the problems when the contest starts
+        if (sc.isUserInRole(WebServer.WEBAPI_ROLE_ADMIN) || sc.isUserInRole(WebServer.WEBAPI_ROLE_JUDGE) || model.getContestTime().isContestStarted()) {
```

Comment:
It seems a bit weird to me that the effect of not having an `else` clause for this `if` statement is that it falls through to the code which formats the JSON and returns `OK` -- meaning that it returns an empty array.  I would think an attempt to read the problems when not Admin and not Judge and not ContestStarted should be (something like) FORBIDDEN.

Is there something in the CLICS spec which indicates an empty array should be returned?  (I looked but couldn't find anything... although at some point in time I do vaguely recall seeing _something_ like that...)